Java code generator generates unit tests that fail when comparing floating point numbers with assertEquals

I do not have all the protos related to the issue checked in. The following example will illustrate the problem.

Let's assume that the service proto has the following definition:

```proto
service TestService {
  rpc TestMethod(TestMethodRequest)
      returns (TestMethodResponse) {
    option (google.api.http) = {
      get: "/v1alpha1/{name=projects/*}"
    };
    // Generates a helper method with two parameters (Strgin name, double fraction)
    option (google.api.method_signature) = "name,fraction";
  }
}

message TestMethodRequest {
  string name = 1 [
    (google.api.field_behavior) = REQUIRED,
  ];
  double fraction = 2;
}

message TestMethodResponse {
  string result = 1;
}
```
The BAZEL build files have Java targets declared similar to this one:
https://github.com/googleapis/googleapis/blob/d9e328eaf790d4e4346fbbf32858160f497a03e0/google/cloud/bigquery/storage/v1beta1/BUILD.bazel#L85-L93

The generated code will have two overloaded methods:
* `public final TestMethodResponse testMethod(TestMethodRequest request);`
* `public final TestMethodResponse testMethod(String name, double fraction);`

The auto-generated unit tests will contain a comparison of the `fraction` floating point parameter with `assertEquals`. That code generates the following error when executing the unit tests:

```
java.lang.AssertionError: Use assertEquals(expected, actual, delta) to compare floating-point numbers
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertEquals(Assert.java:667)
	at org.junit.Assert.assertEquals(Assert.java:656)
...
```

Seems like the Java code generation should avoid using assertEquals for floating point comparisons.

--- a/src/main/java/com/google/api/codegen/transformer/InitCodeTransformer.java
+++ b/src/main/java/com/google/api/codegen/transformer/InitCodeTransformer.java
@@ -207,7 +207,9 @@ public class InitCodeTransformer {
 
       boolean isMap = fieldConfig.getField().isMap();
       boolean isArray = fieldConfig.getField().isRepeated() && !isMap;
-
+      boolean isFloatingPointNumber =
+          fieldConfig.getField().getType().isFloatType()
+              || fieldConfig.getField().getType().isDoubleType();
       TypeModel fieldType = fieldItemTree.getType();
       String messageTypeName = null;
       if (fieldType.isMessage()) {
@@ -221,6 +223,7 @@ public class InitCodeTransformer {
               actualTransformFunction,
               isMap,
               isArray,
+              isFloatingPointNumber,
               getterMethod,
               messageTypeName));
     }
@@ -252,12 +255,14 @@ public class InitCodeTransformer {
       String actualTransformFunction,
       boolean isMap,
       boolean isArray,
+      boolean isFloatingPointNumber,
       String actual,
       String messageTypeName) {
     return ClientTestAssertView.newBuilder()
         .expectedValueIdentifier(expected)
         .isMap(isMap)
         .isArray(isArray)
+        .isFloatingPointNumber(isFloatingPointNumber)
         .expectedValueTransformFunction(expectedTransformFunction)
         .actualValueTransformFunction(actualTransformFunction)
         .actualValueGetter(actual)

--- a/src/main/java/com/google/api/codegen/viewmodel/testing/ClientTestAssertView.java
+++ b/src/main/java/com/google/api/codegen/viewmodel/testing/ClientTestAssertView.java
@@ -27,6 +27,8 @@ public abstract class ClientTestAssertView {
 
   public abstract boolean isArray();
 
+  public abstract boolean isFloatingPointNumber();
+
   @Nullable
   public abstract String expectedValueTransformFunction();
 
@@ -64,6 +66,8 @@ public abstract class ClientTestAssertView {
     public abstract Builder isMap(boolean val);
 
     public abstract Builder isArray(boolean val);
+
+    public abstract Builder isFloatingPointNumber(boolean val);
 
     public abstract Builder expectedValueTransformFunction(String val);
 

--- a/src/main/resources/com/google/api/codegen/java/grpc_test.snip
+++ b/src/main/resources/com/google/api/codegen/java/grpc_test.snip
@@ -302,7 +302,11 @@
   {@test.requestTypeName} actualRequest = ({@test.requestTypeName})actualRequests.get(0);
 
   @join assert : test.asserts
-    Assert.assertEquals({@assertExpectedParam(assert)}, {@assertActualParam(assert)});
+    @if assert.isFloatingPointNumber
+      Assert.assertEquals({@assertExpectedParam(assert)}, {@assertActualParam(assert)}, 0.001);
+    @else
+      Assert.assertEquals({@assertExpectedParam(assert)}, {@assertActualParam(assert)});
+    @end
   @end
   @if test.clientHasDefaultInstance
     Assert.assertTrue(

--- a/src/main/resources/com/google/api/codegen/java/grpc_test.snip
+++ b/src/main/resources/com/google/api/codegen/java/grpc_test.snip
@@ -302,7 +302,7 @@
   {@test.requestTypeName} actualRequest = ({@test.requestTypeName})actualRequests.get(0);
 
   @join assert : test.asserts
-    @if assert.isFloatingPointNumber
+    @if and(assert.isFloatingPointNumber, not(assert.isArray))
       Assert.assertEquals({@assertExpectedParam(assert)}, {@assertActualParam(assert)}, 0.001);
     @else
       Assert.assertEquals({@assertExpectedParam(assert)}, {@assertActualParam(assert)});

--- a/src/test/java/com/google/api/codegen/gapic/testdata/java/java_library.baseline
+++ b/src/test/java/com/google/api/codegen/gapic/testdata/java/java_library.baseline
@@ -15746,8 +15746,8 @@ public class LibraryClientTest {
 
     Assert.assertEquals(requiredSingularInt32, actualRequest.getRequiredSingularInt32());
     Assert.assertEquals(requiredSingularInt64, actualRequest.getRequiredSingularInt64());
-    Assert.assertEquals(requiredSingularFloat, actualRequest.getRequiredSingularFloat());
-    Assert.assertEquals(requiredSingularDouble, actualRequest.getRequiredSingularDouble());
+    Assert.assertEquals(requiredSingularFloat, actualRequest.getRequiredSingularFloat(), 0.001);
+    Assert.assertEquals(requiredSingularDouble, actualRequest.getRequiredSingularDouble(), 0.001);
     Assert.assertEquals(requiredSingularBool, actualRequest.getRequiredSingularBool());
     Assert.assertEquals(requiredSingularEnum, actualRequest.getRequiredSingularEnum());
     Assert.assertEquals(requiredSingularString, actualRequest.getRequiredSingularString());
@@ -15760,8 +15760,8 @@ public class LibraryClientTest {
     Assert.assertEquals(requiredSingularFixed64, actualRequest.getRequiredSingularFixed64());
     Assert.assertEquals(requiredRepeatedInt32, actualRequest.getRequiredRepeatedInt32List());
     Assert.assertEquals(requiredRepeatedInt64, actualRequest.getRequiredRepeatedInt64List());
-    Assert.assertEquals(requiredRepeatedFloat, actualRequest.getRequiredRepeatedFloatList());
-    Assert.assertEquals(requiredRepeatedDouble, actualRequest.getRequiredRepeatedDoubleList());
+    Assert.assertEquals(requiredRepeatedFloat, actualRequest.getRequiredRepeatedFloatList(), 0.001);
+    Assert.assertEquals(requiredRepeatedDouble, actualRequest.getRequiredRepeatedDoubleList(), 0.001);
     Assert.assertEquals(requiredRepeatedBool, actualRequest.getRequiredRepeatedBoolList());
     Assert.assertEquals(requiredRepeatedEnum, actualRequest.getRequiredRepeatedEnumList());
     Assert.assertEquals(requiredRepeatedString, actualRequest.getRequiredRepeatedStringList());
@@ -15807,8 +15807,8 @@ public class LibraryClientTest {
     Assert.assertEquals(requiredRepeatedBytesValue, actualRequest.getRequiredRepeatedBytesValueList());
     Assert.assertEquals(optionalSingularInt32, actualRequest.getOptionalSingularInt32());
     Assert.assertEquals(optionalSingularInt64, actualRequest.getOptionalSingularInt64());
-    Assert.assertEquals(optionalSingularFloat, actualRequest.getOptionalSingularFloat());
-    Assert.assertEquals(optionalSingularDouble, actualRequest.getOptionalSingularDouble());
+    Assert.assertEquals(optionalSingularFloat, actualRequest.getOptionalSingularFloat(), 0.001);
+    Assert.assertEquals(optionalSingularDouble, actualRequest.getOptionalSingularDouble(), 0.001);
     Assert.assertEquals(optionalSingularBool, actualRequest.getOptionalSingularBool());
     Assert.assertEquals(optionalSingularEnum, actualRequest.getOptionalSingularEnum());
     Assert.assertEquals(optionalSingularString, actualRequest.getOptionalSingularString());
@@ -15821,8 +15821,8 @@ public class LibraryClientTest {
     Assert.assertEquals(optionalSingularFixed64, actualRequest.getOptionalSingularFixed64());
     Assert.assertEquals(optionalRepeatedInt32, actualRequest.getOptionalRepeatedInt32List());
     Assert.assertEquals(optionalRepeatedInt64, actualRequest.getOptionalRepeatedInt64List());
-    Assert.assertEquals(optionalRepeatedFloat, actualRequest.getOptionalRepeatedFloatList());
-    Assert.assertEquals(optionalRepeatedDouble, actualRequest.getOptionalRepeatedDoubleList());
+    Assert.assertEquals(optionalRepeatedFloat, actualRequest.getOptionalRepeatedFloatList(), 0.001);
+    Assert.assertEquals(optionalRepeatedDouble, actualRequest.getOptionalRepeatedDoubleList(), 0.001);
     Assert.assertEquals(optionalRepeatedBool, actualRequest.getOptionalRepeatedBoolList());
     Assert.assertEquals(optionalRepeatedEnum, actualRequest.getOptionalRepeatedEnumList());
     Assert.assertEquals(optionalRepeatedString, actualRequest.getOptionalRepeatedStringList());

--- a/src/test/java/com/google/api/codegen/gapic/testdata/java/java_library.baseline
+++ b/src/test/java/com/google/api/codegen/gapic/testdata/java/java_library.baseline
@@ -15760,8 +15760,8 @@ public class LibraryClientTest {
     Assert.assertEquals(requiredSingularFixed64, actualRequest.getRequiredSingularFixed64());
     Assert.assertEquals(requiredRepeatedInt32, actualRequest.getRequiredRepeatedInt32List());
     Assert.assertEquals(requiredRepeatedInt64, actualRequest.getRequiredRepeatedInt64List());
-    Assert.assertEquals(requiredRepeatedFloat, actualRequest.getRequiredRepeatedFloatList(), 0.001);
-    Assert.assertEquals(requiredRepeatedDouble, actualRequest.getRequiredRepeatedDoubleList(), 0.001);
+    Assert.assertEquals(requiredRepeatedFloat, actualRequest.getRequiredRepeatedFloatList());
+    Assert.assertEquals(requiredRepeatedDouble, actualRequest.getRequiredRepeatedDoubleList());
     Assert.assertEquals(requiredRepeatedBool, actualRequest.getRequiredRepeatedBoolList());
     Assert.assertEquals(requiredRepeatedEnum, actualRequest.getRequiredRepeatedEnumList());
     Assert.assertEquals(requiredRepeatedString, actualRequest.getRequiredRepeatedStringList());
@@ -15821,8 +15821,8 @@ public class LibraryClientTest {
     Assert.assertEquals(optionalSingularFixed64, actualRequest.getOptionalSingularFixed64());
     Assert.assertEquals(optionalRepeatedInt32, actualRequest.getOptionalRepeatedInt32List());
     Assert.assertEquals(optionalRepeatedInt64, actualRequest.getOptionalRepeatedInt64List());
-    Assert.assertEquals(optionalRepeatedFloat, actualRequest.getOptionalRepeatedFloatList(), 0.001);
-    Assert.assertEquals(optionalRepeatedDouble, actualRequest.getOptionalRepeatedDoubleList(), 0.001);
+    Assert.assertEquals(optionalRepeatedFloat, actualRequest.getOptionalRepeatedFloatList());
+    Assert.assertEquals(optionalRepeatedDouble, actualRequest.getOptionalRepeatedDoubleList());
     Assert.assertEquals(optionalRepeatedBool, actualRequest.getOptionalRepeatedBoolList());
     Assert.assertEquals(optionalRepeatedEnum, actualRequest.getOptionalRepeatedEnumList());
     Assert.assertEquals(optionalRepeatedString, actualRequest.getOptionalRepeatedStringList());

--- a/src/test/java/com/google/api/codegen/gapic/testdata/java/java_samplegen_config_migration_library.baseline
+++ b/src/test/java/com/google/api/codegen/gapic/testdata/java/java_samplegen_config_migration_library.baseline
@@ -13207,8 +13207,8 @@ public class LibraryClientTest {
     Assert.assertEquals(requiredSingularFixed64, actualRequest.getRequiredSingularFixed64());
     Assert.assertEquals(requiredRepeatedInt32, actualRequest.getRequiredRepeatedInt32List());
     Assert.assertEquals(requiredRepeatedInt64, actualRequest.getRequiredRepeatedInt64List());
-    Assert.assertEquals(requiredRepeatedFloat, actualRequest.getRequiredRepeatedFloatList(), 0.001);
-    Assert.assertEquals(requiredRepeatedDouble, actualRequest.getRequiredRepeatedDoubleList(), 0.001);
+    Assert.assertEquals(requiredRepeatedFloat, actualRequest.getRequiredRepeatedFloatList());
+    Assert.assertEquals(requiredRepeatedDouble, actualRequest.getRequiredRepeatedDoubleList());
     Assert.assertEquals(requiredRepeatedBool, actualRequest.getRequiredRepeatedBoolList());
     Assert.assertEquals(requiredRepeatedEnum, actualRequest.getRequiredRepeatedEnumList());
     Assert.assertEquals(requiredRepeatedString, actualRequest.getRequiredRepeatedStringList());
@@ -13236,8 +13236,8 @@ public class LibraryClientTest {
     Assert.assertEquals(optionalSingularFixed64, actualRequest.getOptionalSingularFixed64());
     Assert.assertEquals(optionalRepeatedInt32, actualRequest.getOptionalRepeatedInt32List());
     Assert.assertEquals(optionalRepeatedInt64, actualRequest.getOptionalRepeatedInt64List());
-    Assert.assertEquals(optionalRepeatedFloat, actualRequest.getOptionalRepeatedFloatList(), 0.001);
-    Assert.assertEquals(optionalRepeatedDouble, actualRequest.getOptionalRepeatedDoubleList(), 0.001);
+    Assert.assertEquals(optionalRepeatedFloat, actualRequest.getOptionalRepeatedFloatList());
+    Assert.assertEquals(optionalRepeatedDouble, actualRequest.getOptionalRepeatedDoubleList());
     Assert.assertEquals(optionalRepeatedBool, actualRequest.getOptionalRepeatedBoolList());
     Assert.assertEquals(optionalRepeatedEnum, actualRequest.getOptionalRepeatedEnumList());
     Assert.assertEquals(optionalRepeatedString, actualRequest.getOptionalRepeatedStringList());

--- a/src/test/java/com/google/api/codegen/gapic/testdata/java/java_samplegen_config_migration_library.baseline
+++ b/src/test/java/com/google/api/codegen/gapic/testdata/java/java_samplegen_config_migration_library.baseline
@@ -13193,8 +13193,8 @@ public class LibraryClientTest {
 
     Assert.assertEquals(requiredSingularInt32, actualRequest.getRequiredSingularInt32());
     Assert.assertEquals(requiredSingularInt64, actualRequest.getRequiredSingularInt64());
-    Assert.assertEquals(requiredSingularFloat, actualRequest.getRequiredSingularFloat());
-    Assert.assertEquals(requiredSingularDouble, actualRequest.getRequiredSingularDouble());
+    Assert.assertEquals(requiredSingularFloat, actualRequest.getRequiredSingularFloat(), 0.001);
+    Assert.assertEquals(requiredSingularDouble, actualRequest.getRequiredSingularDouble(), 0.001);
     Assert.assertEquals(requiredSingularBool, actualRequest.getRequiredSingularBool());
     Assert.assertEquals(requiredSingularEnum, actualRequest.getRequiredSingularEnum());
     Assert.assertEquals(requiredSingularString, actualRequest.getRequiredSingularString());
@@ -13207,8 +13207,8 @@ public class LibraryClientTest {
     Assert.assertEquals(requiredSingularFixed64, actualRequest.getRequiredSingularFixed64());
     Assert.assertEquals(requiredRepeatedInt32, actualRequest.getRequiredRepeatedInt32List());
     Assert.assertEquals(requiredRepeatedInt64, actualRequest.getRequiredRepeatedInt64List());
-    Assert.assertEquals(requiredRepeatedFloat, actualRequest.getRequiredRepeatedFloatList());
-    Assert.assertEquals(requiredRepeatedDouble, actualRequest.getRequiredRepeatedDoubleList());
+    Assert.assertEquals(requiredRepeatedFloat, actualRequest.getRequiredRepeatedFloatList(), 0.001);
+    Assert.assertEquals(requiredRepeatedDouble, actualRequest.getRequiredRepeatedDoubleList(), 0.001);
     Assert.assertEquals(requiredRepeatedBool, actualRequest.getRequiredRepeatedBoolList());
     Assert.assertEquals(requiredRepeatedEnum, actualRequest.getRequiredRepeatedEnumList());
     Assert.assertEquals(requiredRepeatedString, actualRequest.getRequiredRepeatedStringList());
@@ -13222,8 +13222,8 @@ public class LibraryClientTest {
     Assert.assertEquals(requiredMap, actualRequest.getRequiredMapMap());
     Assert.assertEquals(optionalSingularInt32, actualRequest.getOptionalSingularInt32());
     Assert.assertEquals(optionalSingularInt64, actualRequest.getOptionalSingularInt64());
-    Assert.assertEquals(optionalSingularFloat, actualRequest.getOptionalSingularFloat());
-    Assert.assertEquals(optionalSingularDouble, actualRequest.getOptionalSingularDouble());
+    Assert.assertEquals(optionalSingularFloat, actualRequest.getOptionalSingularFloat(), 0.001);
+    Assert.assertEquals(optionalSingularDouble, actualRequest.getOptionalSingularDouble(), 0.001);
     Assert.assertEquals(optionalSingularBool, actualRequest.getOptionalSingularBool());
     Assert.assertEquals(optionalSingularEnum, actualRequest.getOptionalSingularEnum());
     Assert.assertEquals(optionalSingularString, actualRequest.getOptionalSingularString());
@@ -13236,8 +13236,8 @@ public class LibraryClientTest {
     Assert.assertEquals(optionalSingularFixed64, actualRequest.getOptionalSingularFixed64());
     Assert.assertEquals(optionalRepeatedInt32, actualRequest.getOptionalRepeatedInt32List());
     Assert.assertEquals(optionalRepeatedInt64, actualRequest.getOptionalRepeatedInt64List());
-    Assert.assertEquals(optionalRepeatedFloat, actualRequest.getOptionalRepeatedFloatList());
-    Assert.assertEquals(optionalRepeatedDouble, actualRequest.getOptionalRepeatedDoubleList());
+    Assert.assertEquals(optionalRepeatedFloat, actualRequest.getOptionalRepeatedFloatList(), 0.001);
+    Assert.assertEquals(optionalRepeatedDouble, actualRequest.getOptionalRepeatedDoubleList(), 0.001);
     Assert.assertEquals(optionalRepeatedBool, actualRequest.getOptionalRepeatedBoolList());
     Assert.assertEquals(optionalRepeatedEnum, actualRequest.getOptionalRepeatedEnumList());
     Assert.assertEquals(optionalRepeatedString, actualRequest.getOptionalRepeatedStringList());

--- a/src/test/java/com/google/api/codegen/protoannotations/testdata/java_library.baseline
+++ b/src/test/java/com/google/api/codegen/protoannotations/testdata/java_library.baseline
@@ -16562,8 +16562,8 @@ public class LibraryClientTest {
     Assert.assertEquals(requiredSingularFixed64, actualRequest.getRequiredSingularFixed64());
     Assert.assertEquals(requiredRepeatedInt32, actualRequest.getRequiredRepeatedInt32List());
     Assert.assertEquals(requiredRepeatedInt64, actualRequest.getRequiredRepeatedInt64List());
-    Assert.assertEquals(requiredRepeatedFloat, actualRequest.getRequiredRepeatedFloatList(), 0.001);
-    Assert.assertEquals(requiredRepeatedDouble, actualRequest.getRequiredRepeatedDoubleList(), 0.001);
+    Assert.assertEquals(requiredRepeatedFloat, actualRequest.getRequiredRepeatedFloatList());
+    Assert.assertEquals(requiredRepeatedDouble, actualRequest.getRequiredRepeatedDoubleList());
     Assert.assertEquals(requiredRepeatedBool, actualRequest.getRequiredRepeatedBoolList());
     Assert.assertEquals(requiredRepeatedEnum, actualRequest.getRequiredRepeatedEnumList());
     Assert.assertEquals(requiredRepeatedString, actualRequest.getRequiredRepeatedStringList());
@@ -16623,8 +16623,8 @@ public class LibraryClientTest {
     Assert.assertEquals(optionalSingularFixed64, actualRequest.getOptionalSingularFixed64());
     Assert.assertEquals(optionalRepeatedInt32, actualRequest.getOptionalRepeatedInt32List());
     Assert.assertEquals(optionalRepeatedInt64, actualRequest.getOptionalRepeatedInt64List());
-    Assert.assertEquals(optionalRepeatedFloat, actualRequest.getOptionalRepeatedFloatList(), 0.001);
-    Assert.assertEquals(optionalRepeatedDouble, actualRequest.getOptionalRepeatedDoubleList(), 0.001);
+    Assert.assertEquals(optionalRepeatedFloat, actualRequest.getOptionalRepeatedFloatList());
+    Assert.assertEquals(optionalRepeatedDouble, actualRequest.getOptionalRepeatedDoubleList());
     Assert.assertEquals(optionalRepeatedBool, actualRequest.getOptionalRepeatedBoolList());
     Assert.assertEquals(optionalRepeatedEnum, actualRequest.getOptionalRepeatedEnumList());
     Assert.assertEquals(optionalRepeatedString, actualRequest.getOptionalRepeatedStringList());

--- a/src/test/java/com/google/api/codegen/protoannotations/testdata/java_library.baseline
+++ b/src/test/java/com/google/api/codegen/protoannotations/testdata/java_library.baseline
@@ -16548,8 +16548,8 @@ public class LibraryClientTest {
 
     Assert.assertEquals(requiredSingularInt32, actualRequest.getRequiredSingularInt32());
     Assert.assertEquals(requiredSingularInt64, actualRequest.getRequiredSingularInt64());
-    Assert.assertEquals(requiredSingularFloat, actualRequest.getRequiredSingularFloat());
-    Assert.assertEquals(requiredSingularDouble, actualRequest.getRequiredSingularDouble());
+    Assert.assertEquals(requiredSingularFloat, actualRequest.getRequiredSingularFloat(), 0.001);
+    Assert.assertEquals(requiredSingularDouble, actualRequest.getRequiredSingularDouble(), 0.001);
     Assert.assertEquals(requiredSingularBool, actualRequest.getRequiredSingularBool());
     Assert.assertEquals(requiredSingularEnum, actualRequest.getRequiredSingularEnum());
     Assert.assertEquals(requiredSingularString, actualRequest.getRequiredSingularString());
@@ -16562,8 +16562,8 @@ public class LibraryClientTest {
     Assert.assertEquals(requiredSingularFixed64, actualRequest.getRequiredSingularFixed64());
     Assert.assertEquals(requiredRepeatedInt32, actualRequest.getRequiredRepeatedInt32List());
     Assert.assertEquals(requiredRepeatedInt64, actualRequest.getRequiredRepeatedInt64List());
-    Assert.assertEquals(requiredRepeatedFloat, actualRequest.getRequiredRepeatedFloatList());
-    Assert.assertEquals(requiredRepeatedDouble, actualRequest.getRequiredRepeatedDoubleList());
+    Assert.assertEquals(requiredRepeatedFloat, actualRequest.getRequiredRepeatedFloatList(), 0.001);
+    Assert.assertEquals(requiredRepeatedDouble, actualRequest.getRequiredRepeatedDoubleList(), 0.001);
     Assert.assertEquals(requiredRepeatedBool, actualRequest.getRequiredRepeatedBoolList());
     Assert.assertEquals(requiredRepeatedEnum, actualRequest.getRequiredRepeatedEnumList());
     Assert.assertEquals(requiredRepeatedString, actualRequest.getRequiredRepeatedStringList());
@@ -16609,8 +16609,8 @@ public class LibraryClientTest {
     Assert.assertEquals(requiredRepeatedBytesValue, actualRequest.getRequiredRepeatedBytesValueList());
     Assert.assertEquals(optionalSingularInt32, actualRequest.getOptionalSingularInt32());
     Assert.assertEquals(optionalSingularInt64, actualRequest.getOptionalSingularInt64());
-    Assert.assertEquals(optionalSingularFloat, actualRequest.getOptionalSingularFloat());
-    Assert.assertEquals(optionalSingularDouble, actualRequest.getOptionalSingularDouble());
+    Assert.assertEquals(optionalSingularFloat, actualRequest.getOptionalSingularFloat(), 0.001);
+    Assert.assertEquals(optionalSingularDouble, actualRequest.getOptionalSingularDouble(), 0.001);
     Assert.assertEquals(optionalSingularBool, actualRequest.getOptionalSingularBool());
     Assert.assertEquals(optionalSingularEnum, actualRequest.getOptionalSingularEnum());
     Assert.assertEquals(optionalSingularString, actualRequest.getOptionalSingularString());
@@ -16623,8 +16623,8 @@ public class LibraryClientTest {
     Assert.assertEquals(optionalSingularFixed64, actualRequest.getOptionalSingularFixed64());
     Assert.assertEquals(optionalRepeatedInt32, actualRequest.getOptionalRepeatedInt32List());
     Assert.assertEquals(optionalRepeatedInt64, actualRequest.getOptionalRepeatedInt64List());
-    Assert.assertEquals(optionalRepeatedFloat, actualRequest.getOptionalRepeatedFloatList());
-    Assert.assertEquals(optionalRepeatedDouble, actualRequest.getOptionalRepeatedDoubleList());
+    Assert.assertEquals(optionalRepeatedFloat, actualRequest.getOptionalRepeatedFloatList(), 0.001);
+    Assert.assertEquals(optionalRepeatedDouble, actualRequest.getOptionalRepeatedDoubleList(), 0.001);
     Assert.assertEquals(optionalRepeatedBool, actualRequest.getOptionalRepeatedBoolList());
     Assert.assertEquals(optionalRepeatedEnum, actualRequest.getOptionalRepeatedEnumList());
     Assert.assertEquals(optionalRepeatedString, actualRequest.getOptionalRepeatedStringList());

--- a/src/test/java/com/google/api/codegen/protoannotations/testdata/java_library_no_gapic_config.baseline
+++ b/src/test/java/com/google/api/codegen/protoannotations/testdata/java_library_no_gapic_config.baseline
@@ -11759,8 +11759,8 @@ public class LibraryServiceClientTest {
 
     Assert.assertEquals(requiredSingularInt32, actualRequest.getRequiredSingularInt32());
     Assert.assertEquals(requiredSingularInt64, actualRequest.getRequiredSingularInt64());
-    Assert.assertEquals(requiredSingularFloat, actualRequest.getRequiredSingularFloat());
-    Assert.assertEquals(requiredSingularDouble, actualRequest.getRequiredSingularDouble());
+    Assert.assertEquals(requiredSingularFloat, actualRequest.getRequiredSingularFloat(), 0.001);
+    Assert.assertEquals(requiredSingularDouble, actualRequest.getRequiredSingularDouble(), 0.001);
     Assert.assertEquals(requiredSingularBool, actualRequest.getRequiredSingularBool());
     Assert.assertEquals(requiredSingularEnum, actualRequest.getRequiredSingularEnum());
     Assert.assertEquals(requiredSingularString, actualRequest.getRequiredSingularString());
@@ -11773,8 +11773,8 @@ public class LibraryServiceClientTest {
     Assert.assertEquals(requiredSingularFixed64, actualRequest.getRequiredSingularFixed64());
     Assert.assertEquals(requiredRepeatedInt32, actualRequest.getRequiredRepeatedInt32List());
     Assert.assertEquals(requiredRepeatedInt64, actualRequest.getRequiredRepeatedInt64List());
-    Assert.assertEquals(requiredRepeatedFloat, actualRequest.getRequiredRepeatedFloatList());
-    Assert.assertEquals(requiredRepeatedDouble, actualRequest.getRequiredRepeatedDoubleList());
+    Assert.assertEquals(requiredRepeatedFloat, actualRequest.getRequiredRepeatedFloatList(), 0.001);
+    Assert.assertEquals(requiredRepeatedDouble, actualRequest.getRequiredRepeatedDoubleList(), 0.001);
     Assert.assertEquals(requiredRepeatedBool, actualRequest.getRequiredRepeatedBoolList());
     Assert.assertEquals(requiredRepeatedEnum, actualRequest.getRequiredRepeatedEnumList());
     Assert.assertEquals(requiredRepeatedString, actualRequest.getRequiredRepeatedStringList());
@@ -11820,8 +11820,8 @@ public class LibraryServiceClientTest {
     Assert.assertEquals(requiredRepeatedBytesValue, actualRequest.getRequiredRepeatedBytesValueList());
     Assert.assertEquals(optionalSingularInt32, actualRequest.getOptionalSingularInt32());
     Assert.assertEquals(optionalSingularInt64, actualRequest.getOptionalSingularInt64());
-    Assert.assertEquals(optionalSingularFloat, actualRequest.getOptionalSingularFloat());
-    Assert.assertEquals(optionalSingularDouble, actualRequest.getOptionalSingularDouble());
+    Assert.assertEquals(optionalSingularFloat, actualRequest.getOptionalSingularFloat(), 0.001);
+    Assert.assertEquals(optionalSingularDouble, actualRequest.getOptionalSingularDouble(), 0.001);
     Assert.assertEquals(optionalSingularBool, actualRequest.getOptionalSingularBool());
     Assert.assertEquals(optionalSingularEnum, actualRequest.getOptionalSingularEnum());
     Assert.assertEquals(optionalSingularString, actualRequest.getOptionalSingularString());
@@ -11834,8 +11834,8 @@ public class LibraryServiceClientTest {
     Assert.assertEquals(optionalSingularFixed64, actualRequest.getOptionalSingularFixed64());
     Assert.assertEquals(optionalRepeatedInt32, actualRequest.getOptionalRepeatedInt32List());
     Assert.assertEquals(optionalRepeatedInt64, actualRequest.getOptionalRepeatedInt64List());
-    Assert.assertEquals(optionalRepeatedFloat, actualRequest.getOptionalRepeatedFloatList());
-    Assert.assertEquals(optionalRepeatedDouble, actualRequest.getOptionalRepeatedDoubleList());
+    Assert.assertEquals(optionalRepeatedFloat, actualRequest.getOptionalRepeatedFloatList(), 0.001);
+    Assert.assertEquals(optionalRepeatedDouble, actualRequest.getOptionalRepeatedDoubleList(), 0.001);
     Assert.assertEquals(optionalRepeatedBool, actualRequest.getOptionalRepeatedBoolList());
     Assert.assertEquals(optionalRepeatedEnum, actualRequest.getOptionalRepeatedEnumList());
     Assert.assertEquals(optionalRepeatedString, actualRequest.getOptionalRepeatedStringList());

--- a/src/test/java/com/google/api/codegen/protoannotations/testdata/java_library_no_gapic_config.baseline
+++ b/src/test/java/com/google/api/codegen/protoannotations/testdata/java_library_no_gapic_config.baseline
@@ -11773,8 +11773,8 @@ public class LibraryServiceClientTest {
     Assert.assertEquals(requiredSingularFixed64, actualRequest.getRequiredSingularFixed64());
     Assert.assertEquals(requiredRepeatedInt32, actualRequest.getRequiredRepeatedInt32List());
     Assert.assertEquals(requiredRepeatedInt64, actualRequest.getRequiredRepeatedInt64List());
-    Assert.assertEquals(requiredRepeatedFloat, actualRequest.getRequiredRepeatedFloatList(), 0.001);
-    Assert.assertEquals(requiredRepeatedDouble, actualRequest.getRequiredRepeatedDoubleList(), 0.001);
+    Assert.assertEquals(requiredRepeatedFloat, actualRequest.getRequiredRepeatedFloatList());
+    Assert.assertEquals(requiredRepeatedDouble, actualRequest.getRequiredRepeatedDoubleList());
     Assert.assertEquals(requiredRepeatedBool, actualRequest.getRequiredRepeatedBoolList());
     Assert.assertEquals(requiredRepeatedEnum, actualRequest.getRequiredRepeatedEnumList());
     Assert.assertEquals(requiredRepeatedString, actualRequest.getRequiredRepeatedStringList());
@@ -11834,8 +11834,8 @@ public class LibraryServiceClientTest {
     Assert.assertEquals(optionalSingularFixed64, actualRequest.getOptionalSingularFixed64());
     Assert.assertEquals(optionalRepeatedInt32, actualRequest.getOptionalRepeatedInt32List());
     Assert.assertEquals(optionalRepeatedInt64, actualRequest.getOptionalRepeatedInt64List());
-    Assert.assertEquals(optionalRepeatedFloat, actualRequest.getOptionalRepeatedFloatList(), 0.001);
-    Assert.assertEquals(optionalRepeatedDouble, actualRequest.getOptionalRepeatedDoubleList(), 0.001);
+    Assert.assertEquals(optionalRepeatedFloat, actualRequest.getOptionalRepeatedFloatList());
+    Assert.assertEquals(optionalRepeatedDouble, actualRequest.getOptionalRepeatedDoubleList());
     Assert.assertEquals(optionalRepeatedBool, actualRequest.getOptionalRepeatedBoolList());
     Assert.assertEquals(optionalRepeatedEnum, actualRequest.getOptionalRepeatedEnumList());
     Assert.assertEquals(optionalRepeatedString, actualRequest.getOptionalRepeatedStringList());

--- a/src/test/java/com/google/api/codegen/protoannotations/testdata/java_library_with_grpc_service_config.baseline
+++ b/src/test/java/com/google/api/codegen/protoannotations/testdata/java_library_with_grpc_service_config.baseline
@@ -16566,8 +16566,8 @@ public class LibraryClientTest {
     Assert.assertEquals(requiredSingularFixed64, actualRequest.getRequiredSingularFixed64());
     Assert.assertEquals(requiredRepeatedInt32, actualRequest.getRequiredRepeatedInt32List());
     Assert.assertEquals(requiredRepeatedInt64, actualRequest.getRequiredRepeatedInt64List());
-    Assert.assertEquals(requiredRepeatedFloat, actualRequest.getRequiredRepeatedFloatList(), 0.001);
-    Assert.assertEquals(requiredRepeatedDouble, actualRequest.getRequiredRepeatedDoubleList(), 0.001);
+    Assert.assertEquals(requiredRepeatedFloat, actualRequest.getRequiredRepeatedFloatList());
+    Assert.assertEquals(requiredRepeatedDouble, actualRequest.getRequiredRepeatedDoubleList());
     Assert.assertEquals(requiredRepeatedBool, actualRequest.getRequiredRepeatedBoolList());
     Assert.assertEquals(requiredRepeatedEnum, actualRequest.getRequiredRepeatedEnumList());
     Assert.assertEquals(requiredRepeatedString, actualRequest.getRequiredRepeatedStringList());
@@ -16627,8 +16627,8 @@ public class LibraryClientTest {
     Assert.assertEquals(optionalSingularFixed64, actualRequest.getOptionalSingularFixed64());
     Assert.assertEquals(optionalRepeatedInt32, actualRequest.getOptionalRepeatedInt32List());
     Assert.assertEquals(optionalRepeatedInt64, actualRequest.getOptionalRepeatedInt64List());
-    Assert.assertEquals(optionalRepeatedFloat, actualRequest.getOptionalRepeatedFloatList(), 0.001);
-    Assert.assertEquals(optionalRepeatedDouble, actualRequest.getOptionalRepeatedDoubleList(), 0.001);
+    Assert.assertEquals(optionalRepeatedFloat, actualRequest.getOptionalRepeatedFloatList());
+    Assert.assertEquals(optionalRepeatedDouble, actualRequest.getOptionalRepeatedDoubleList());
     Assert.assertEquals(optionalRepeatedBool, actualRequest.getOptionalRepeatedBoolList());
     Assert.assertEquals(optionalRepeatedEnum, actualRequest.getOptionalRepeatedEnumList());
     Assert.assertEquals(optionalRepeatedString, actualRequest.getOptionalRepeatedStringList());

--- a/src/test/java/com/google/api/codegen/protoannotations/testdata/java_library_with_grpc_service_config.baseline
+++ b/src/test/java/com/google/api/codegen/protoannotations/testdata/java_library_with_grpc_service_config.baseline
@@ -16552,8 +16552,8 @@ public class LibraryClientTest {
 
     Assert.assertEquals(requiredSingularInt32, actualRequest.getRequiredSingularInt32());
     Assert.assertEquals(requiredSingularInt64, actualRequest.getRequiredSingularInt64());
-    Assert.assertEquals(requiredSingularFloat, actualRequest.getRequiredSingularFloat());
-    Assert.assertEquals(requiredSingularDouble, actualRequest.getRequiredSingularDouble());
+    Assert.assertEquals(requiredSingularFloat, actualRequest.getRequiredSingularFloat(), 0.001);
+    Assert.assertEquals(requiredSingularDouble, actualRequest.getRequiredSingularDouble(), 0.001);
     Assert.assertEquals(requiredSingularBool, actualRequest.getRequiredSingularBool());
     Assert.assertEquals(requiredSingularEnum, actualRequest.getRequiredSingularEnum());
     Assert.assertEquals(requiredSingularString, actualRequest.getRequiredSingularString());
@@ -16566,8 +16566,8 @@ public class LibraryClientTest {
     Assert.assertEquals(requiredSingularFixed64, actualRequest.getRequiredSingularFixed64());
     Assert.assertEquals(requiredRepeatedInt32, actualRequest.getRequiredRepeatedInt32List());
     Assert.assertEquals(requiredRepeatedInt64, actualRequest.getRequiredRepeatedInt64List());
-    Assert.assertEquals(requiredRepeatedFloat, actualRequest.getRequiredRepeatedFloatList());
-    Assert.assertEquals(requiredRepeatedDouble, actualRequest.getRequiredRepeatedDoubleList());
+    Assert.assertEquals(requiredRepeatedFloat, actualRequest.getRequiredRepeatedFloatList(), 0.001);
+    Assert.assertEquals(requiredRepeatedDouble, actualRequest.getRequiredRepeatedDoubleList(), 0.001);
     Assert.assertEquals(requiredRepeatedBool, actualRequest.getRequiredRepeatedBoolList());
     Assert.assertEquals(requiredRepeatedEnum, actualRequest.getRequiredRepeatedEnumList());
     Assert.assertEquals(requiredRepeatedString, actualRequest.getRequiredRepeatedStringList());
@@ -16613,8 +16613,8 @@ public class LibraryClientTest {
     Assert.assertEquals(requiredRepeatedBytesValue, actualRequest.getRequiredRepeatedBytesValueList());
     Assert.assertEquals(optionalSingularInt32, actualRequest.getOptionalSingularInt32());
     Assert.assertEquals(optionalSingularInt64, actualRequest.getOptionalSingularInt64());
-    Assert.assertEquals(optionalSingularFloat, actualRequest.getOptionalSingularFloat());
-    Assert.assertEquals(optionalSingularDouble, actualRequest.getOptionalSingularDouble());
+    Assert.assertEquals(optionalSingularFloat, actualRequest.getOptionalSingularFloat(), 0.001);
+    Assert.assertEquals(optionalSingularDouble, actualRequest.getOptionalSingularDouble(), 0.001);
     Assert.assertEquals(optionalSingularBool, actualRequest.getOptionalSingularBool());
     Assert.assertEquals(optionalSingularEnum, actualRequest.getOptionalSingularEnum());
     Assert.assertEquals(optionalSingularString, actualRequest.getOptionalSingularString());
@@ -16627,8 +16627,8 @@ public class LibraryClientTest {
     Assert.assertEquals(optionalSingularFixed64, actualRequest.getOptionalSingularFixed64());
     Assert.assertEquals(optionalRepeatedInt32, actualRequest.getOptionalRepeatedInt32List());
     Assert.assertEquals(optionalRepeatedInt64, actualRequest.getOptionalRepeatedInt64List());
-    Assert.assertEquals(optionalRepeatedFloat, actualRequest.getOptionalRepeatedFloatList());
-    Assert.assertEquals(optionalRepeatedDouble, actualRequest.getOptionalRepeatedDoubleList());
+    Assert.assertEquals(optionalRepeatedFloat, actualRequest.getOptionalRepeatedFloatList(), 0.001);
+    Assert.assertEquals(optionalRepeatedDouble, actualRequest.getOptionalRepeatedDoubleList(), 0.001);
     Assert.assertEquals(optionalRepeatedBool, actualRequest.getOptionalRepeatedBoolList());
     Assert.assertEquals(optionalRepeatedEnum, actualRequest.getOptionalRepeatedEnumList());
     Assert.assertEquals(optionalRepeatedString, actualRequest.getOptionalRepeatedStringList());

